src/tir/transforms/make_packed_api.cc

```diff
@@ -41,6 +41,56 @@
 namespace tvm {
 namespace tir {
 
+class ReturnRewriter : public StmtMutator {
+ public:
+  explicit ReturnRewriter(Var ret_var, Var ret_tcode) : ret_var_(ret_var), ret_tcode_(ret_tcode) {}
+
+  Stmt VisitStmt_(const EvaluateNode* node) override {
+    Stmt ret = StmtMutator::VisitStmt_(node);
+    const EvaluateNode* eval = ret.as<EvaluateNode>();
+    CHECK(eval);
+    if (const CallNode* call = eval->value.as<CallNode>()) {
+      if (call->op.same_as(builtin::ret())) {
+        CHECK_EQ(call->args.size(), 1);
```

Comment:
ICHECK_EQ with error message

Reply:
This is a code invariant so changing to ICHECK should suffice

Reply:
See above

Reply:
Thanks @tkonolige , I now agree that in this particular case i agree that having an error message like "ret expect a single argument" would be helpful.

Reply:
Things like this is quite easy to understand by eyeball the error throwing, I don't think error message is necessary for every place or would you like to add error message for every place like here? https://github.com/apache/tvm/blob/main/src/relay/op/nn/pad.cc#L58-L59

Reply:
When you've written the code thinks may appear simple to you, but a new user or outside maybe be very confused.

Yes, I think we should add error messages in all those places.

Reply:
I think the rule of thumb is whether or not a user might hit the error, and we should look at the context of the check. 

Notably, we encourage putting ICHECK deliberately in many places and the ease of adding checks quickly facilitate development. Notably, if it is an ICHECK, it usually means it is not a user facing error. We need to consider the context in each case 

In this particular case, an error could be hit when the call is incorrectly constructed and the IR being invalid. Which is a rare case, but user could hit it in certain cases when invalid IR is constructed , so I would say an error message would be helpful. Although we should also think to which point is helpful.